### ISSUE TITLE
FED Calculations

### ISSUE THREAD

**Issue by gforney** [firemodels/fds#1734]
```
Please complete the following:

Application Version: FDS6
SVN Revision Number: Beta
Compile Date: 2/11/2012

Description of enhancement.  Be as descriptive as possible including if
needed equations and units.

#1 The issue is with the manual and being consistent.  In section 16.9.10 of the User
Guide FDS6, the FED the units are in minutes.  However, previous versions of the manual
used seconds which also is consistent with the units used within program itself.  I
would suggest that you use seconds for this section as has been done in the other parts
of the manual.  Section 6.4.5 is the only other location I found minutes mentioned
in the User Guide.

#2 You might consider using (The SFPE Handbook, 4th Edition) Purser's equation 7 directly
then allow the user to specify the breathing rate (V) and exposure does to incapacitation
(D) thus making the FED analysis more generic.  The current form of equation 16.19
in the User Guide assumes that the occupant is doing "Light work" and cannot be changed
within the input file.  Therefore, to deal with a sleeping occupant or a "Heavy work"
occupant the user must do some post processing. I would suggest you allow the user
to change D & V and set the defaults for D=30 COHb and V=25 L/min.

#3 Consider adding an FED to account for the thermal effects.  Page 2-145 of  "The
SFPE Handbook" (4th Edition) outline the methodology for doing this.  All of the pieces
are already in FDS, it would simply be helpful to include Equations 33, 34, and 35
into FDS6.  This is also described in ISO13571

Keep up the good work, we appreciate all efforts that go into keeping FDS going. 

Regards,

Charley
```

Original issue reported on code.google.com by `nzcharley` on 2012-11-07 01:43:23

**Comment by gforney** [comment]
```
Thanks, Charley. I'll ask Simo to look into this.
```

Original issue reported on code.google.com by `mcgratta` on 2012-11-07 13:27:54

**Comment by gforney** [comment]
```
Well, there is something done for the point #2 already.
The point #1 could be easily changed in the manual if
wanted, but RMV = "Respiratory minute volume" so the 
common units are minutes here. Point #3 is on my To Do
list, so that will be added at some sunny day.

FUNCTION FED(Y_IN,RSUM,FED_ACTIVITY)

So, FED_ACTIVITY is added:

!                at rest           light work(default) heavy work
DATA CO_FED_FAC /0.70486250E-5_EB, 2.7641667E-5_EB,    8.2925E-5_EB/

! All equations from D.A. Purser, Sec. 2, Chap. 6, SFPE Handbook, 4th Ed.
! Note: Purser uses minutes, here dt is in seconds. Conversion at the end of the function.
! Total FED dose:
! FED_dose = (FED_LCO + FED_LCN + FED_LNOx + FLD_irr)*FED_VCO2 + FED_LO2;


! Carbon monoxide (CO)
!          at rest    light work heavy work
! RMV_FED /8.5_EB,    25.0_EB,   50.0_EB /
! D_FED   /40.0_EB,   30.0_EB,   20.0_EB /
! RMV/D   /0.2125_EB, 0.8333_EB, 2.5_EB/ 
!

The FED_ACTIVITY is a PROP namelist entry, below are some examples:

&SLCF PBY= 1.5, QUANTITY='INCAPACITATION TIME', PROP_ID='Prop_fed_act'/
&PROP ID='Prop_fed_act', FED_ACTIVITY=2 /

&DEVC XYZ = 1.5,1.5,1.7, QUANTITY = 'FED', ID='Fed', PROP_ID='Prop_fed_act' /
&DEVC XYZ = 1.5,1.5,1.7, QUANTITY = 'INCAPACITATION TIME', ID='T_inc', PROP_ID='Prop_fed_act'
/

So it works for DEVC and SLCF outputs. And there is an "INCAPACITATION TIME"
output quantity added also. It is equal the inverse of the integrand in the
FED formula. Units are minutes. So, it tells how long a human could stand
at this place if the concentrations would remain constant (and the dose
is initially zero at this time point).

These are not yet in the FDS manual, one should add verification tests
to the verification suite.

The evacuation uses FED_ACTIVITY that is given on the PERS namelist.
(Last one read in is used if given on many PERS namelists.)

The default is FED_ACTIVITY = 2 (light work)
1 = at rest, 2 = light work, 3 = heavy work.
Light work is the default also for DEVC and SLCF.

TimoK

```

Original issue reported on code.google.com by `tkorhon1` on 2012-11-13 13:40:55

**Comment by gforney** [comment]
```
Closed means => points #2 and #3 are in Issue 1502 already
and point #1 is "no fix" (minutes it shall be).

Well, this is more or less the same Issue than Issue 1502.
Just the point #1, i.e., the FDS manual has minutes for FED
units. Should the manual have seconds? The equations in the
source code seem to be minutes (numerical data) and at the
end of the fed function dt=min => dt=s transformation is done,
i.e., fed_integrand is divided by 60. 

I would say that one should use dt=minute for the equations in
the manual, because the SFPE and other references are using
minutes. This way the constants will have same numerical 
values and the reader should easily recognize different
constants.

So, I change this issue closed, the #3 point is in the issue 1502.

TimoK

```

Original issue reported on code.google.com by `tkorhon1` on 2012-11-14 08:36:39
